### DISCUSSION TITLE
Multiple Path parameters/variables using decorator function (Get all the Path parameters as a single function argument)

### DISCUSSION THREAD

**Discussion by vsnag** [fastapi/fastapi#9435] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# Code copy-pasted from https://stackoverflow.com/a/69200232/4126092
from fastapi import FastAPI

app = FastAPI()


domains = {"architecture": ["building","floor"], "vehicle": ["engine","wheels"]}

def set_route(domain,entity):
    url = "/{}/entity/{}/{{id}}".format(domain,entity)
    @app.get(url)
    async def read_users(id: int):
        return(f"Users of the {domain} {entity} #{id}")

for domain, entities in domains.items():
    for entity in entities:
        set_route(domain,entity)
```

### Description

Inspired by this SO post answer [fastapi-dynamic-multiple-path-parameters](https://stackoverflow.com/a/69200232/4126092). I want to extend that concept to contain multiple path parameters/variables. The url will look something like `/architecture/{building}/{block}/{floor}/{apartment}/door-type`  with a combination of multiple dynamic & static path parameters. 
The problem that I am hit with is adding multiple parameters to `read_users` function dynamically based on the path construction. I could use python's `exec` to convert a formatted python string to python code such that the dynamic number of arguments can be inserted to the `read_user` function. But `exec` approach looks very costly. 
Is there a way to handle this problem in fastapi? 
I am fine to implement a middle-ware as well. 
The [path-convertor](https://fastapi.tiangolo.com/tutorial/path-params/#path-convertor) documentation is talking file paths, my problem is **not** related to file paths.

### Operating System

Linux

### Operating System Details

Ubuntu-20.04

### FastAPI Version

0.95.1

### Python Version

3.11.3

**Comment by chrisK824** [comment]
Hi @vsnag , I found that question in some other research but this seems interesting and could be useful

However, can you explain a bit better this part of the question please? 

: The problem that I am hit with is adding multiple parameters to read_users function dynamically based on the path construction.

You are presenting one hypothetical path in your question with some path parameters. How an alternative one would look like with "dynamic" number of path parameters? Under which rule would the path,and hence number of parameters, be defined also in such case? Also, would the function under the route differ as well?

Maybe the cases don't even worth finding such a solution in the first place?

If you have already found another approach than exec, please share. 

P.S Dope profile pic 😋
